src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -16,11 +15,14 @@
 public class DefaultSinkRecordConverter implements SinkRecordConverter {
 
 	private DocumentWriteOperationBuilder documentWriteOperationBuilder;
-
 	private Format format;
 	private String mimeType;
-
+	private Boolean addTopicToCollections = true; 
+	
 	public DefaultSinkRecordConverter(Map<String, String> kafkaConfig) {
+		
+		addTopicToCollections = Boolean.parseBoolean((kafkaConfig.get(MarkLogicSinkConfig.DOCUMENT_COLLECTIONS_ADD_TOPIC)).trim());
```

Comment:
Calling "trim" on this will throw a null-pointer exception if the property is not specified or has no value.

Reply:
Yeah, but there is always a default value false, even if it is not set. The trim is too help with unplanned spaces around the value .

Reply:
If the property isn't specified - which is currently the case in marklogic-sink.properties - then the following error is thrown:

```
[2019-12-04 10:52:41,600] ERROR WorkerSinkTask{id=marklogic-sink-0} Task threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:177)
java.lang.NullPointerException
        at com.marklogic.kafka.connect.sink.DefaultSinkRecordConverter.<init>(DefaultSinkRecordConverter.java:25)
```

I recommend for defense purposes, just follow the pattern already in place in that method - i.e.

```
val = kafkaConfig.get(MarkLogicSinkConfig.DOCUMENT_COLLECTIONS_ADD_TOPIC);
if (val != null && val.trim().length() > 0) {
  addTopicToCollections = Boolean.parseBoolean(val.trim());
}
```

I think that's actually more defensive than what's needed - I think you just need "val != null". Either way is fine then.